### PR TITLE
Tweaks timestop so mobs affected by it are more obvious

### PR DESCRIPTION
**Pull request by PKPenguin321**
It will change your color to a sepia brown upon entering it, and reset it once you exit.

Not sure if the code works, knowing me it probably doesn't. Didn't know exact syntax to add args to a proc or if the oldColor var would stay persistent through the if and for statements.

### CODE REVIEW

**Comment by GunHog** [comment]
If you do not know if it works, why not test it first?

**Comment by PKPenguin321** [comment]
Inbuilt editor is easier to set up, and if this doesn't work then I don't know the right way to do it anyways.

**Comment by ExcessiveUseOfCobblestone** [comment]
the code works if you don't test it gun_hog ;-)

File: code/modules/research/xenobiology/xenobiology.dm [hunk 1]
```diff
@@ -581,20 +581,22 @@
 					H.AIStatus = AI_OFF
 					H.LoseTarget()
 				stopped_atoms |= M
+				var oldColor = M.color
+				M.color = "#704214"//sepia brown
```
Comment (coiax) [inline code/modules/research/xenobiology/xenobiology.dm]: We don't use the color var anymore, look at elsewhere in the code where it is done, like the Ratvar immunity spell.

**Comment by MrPerson** [comment]
Wasn't the original idea to make them inverted in color?

File: code/modules/research/xenobiology/xenobiology.dm (hunk 1 above)
Comment (ChangelingRain) [inline code/modules/research/xenobiology/xenobiology.dm]: replace this with M.add_atom_colour("#704214", TEMPORARY_COLOUR_PRIORITY)

File: code/modules/research/xenobiology/xenobiology.dm [hunk 2]
```diff
@@ -603,12 +605,13 @@
...
 
 
-/obj/effect/timestop/proc/unfreeze_mob(mob/living/M)
+/obj/effect/timestop/proc/unfreeze_mob(mob/living/M, var/colorToSet as text)
 	M.AdjustStunned(-10, 1, 1)
 	M.anchored = 0
 	if(ishostile(M))
 		var/mob/living/simple_animal/hostile/H = M
 		H.AIStatus = initial(H.AIStatus)
+	M.color = colorToSet
```
Comment (ChangelingRain) [inline code/modules/research/xenobiology/xenobiology.dm]: Replace this with M.remove_atom_colour(TEMPORARY_COLOUR_PRIORITY, "#704214")

**Comment by coiax** [comment]
image
Negative Zone!

**Comment by PKPenguin321** [comment]
@MrPerson that would be cool but wouldnt it require them to have a set color in the first place in order to invert the value? or youd have to do some kind of weird inverted color overlay meme

File: code/modules/research/xenobiology/xenobiology.dm (hunk 2 above)
Comment (PKPenguin321) [inline code/modules/research/xenobiology/xenobiology.dm]: i guess this means i dont need the new arg

**Comment by MrPerson** [comment]
set the color to list(-1,0,0,0, 0,-1,0,0, 0,0,-1,0, 0,0,0,1, 1,1,1,0) to invert it

**Comment by Kromgar** [comment]
TOKI WO TOMARE

**Comment by PKPenguin321** [comment]
gonna do some testing with this one sec

**Comment by PKPenguin321** [comment]
does the color proc thingy not apply to turfs or something? im trying to invert the colors of every atom the time stop touches but turfs just turn solid black, pretty broken

**Comment by PKPenguin321** [comment]
yeah now it's just all kinds of fucked, inverted colors dont wear off (i'm guessing neither do regular sepia overlays) and half the shit just turns black

neat idea but i am bad

**Pull request closed without merging**